python/inner-write-structural-prior-CV.py

```diff
@@ -168,7 +172,7 @@
 # write to output files
 os.makedirs(os.path.dirname(args.ranks_file), exist_ok=True)
 rank_df.to_csv(args.ranks_file, index=False,
-               compression='gzip')
+               compression='gzip' if args.ranks_file.endswith('.gz') else None)
 os.makedirs(os.path.dirname(args.tc_file), exist_ok=True)
 tc_df.to_csv(args.tc_file, index=False,
-             compression='gzip')
+             compression='gzip' if args.tc_file.endswith('.gz') else None)
```

Comment:
It's hard to do a deterministic checksum on a `gz` file (what if the compression parameters change?) so I've made compression optional. Doesn't hurt since these csv files aren't that huge anyway.